File: socialuni-center-web/src/main/java/com/socialuni/center/web/feignAPI/SocialuniCosAPI.java
package com.socialuni.center.web.feignAPI;

import com.socialuni.social.api.model.ResultRO;
import com.socialuni.center.web.model.RO.app.SocialCosAuthRO;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@RequestMapping("cos")
@FeignClient(name = "cos", url = "${socialuni.server-url:https://api.socialuni.cn}")
public interface SocialuniCosAPI {
    @PostMapping("getCosAuth")
    ResultRO<SocialCosAuthRO> getCosAuth();
}

File: socialuni-js/src/socialuni/api/socialuni/SocialuniCosAPI.ts
import request from '../../plugins/http/request'
import CosAuthRO from '../../model/cos/CosAuthRO'

export default class SocialuniCosAPI {
  static getCosAuthorizationAPI () {
    return request.post<CosAuthRO>('socialuni/cos/getCosAuth')
  }
}
